### PR TITLE
Parameterize the dataplane deployment name

### PR DESCRIPTION
**Pull request by fao89**

### CODE REVIEW

File: scripts/gen-edpm-kustomize.sh [hunk 1]
```diff
@@ -48,6 +48,12 @@ labels:
   - pairs:
       created-by: install_yamls
 patches:
+- target:
+    kind: OpenStackDataPlaneDeployment
+  patch: |-
+    - op: replace
+      path: /metadata/name
+      value: ${DEPLOYMENT_NAME:-"edpm-deployment"}
```
Comment (abays) [inline scripts/gen-edpm-kustomize.sh:56]: Would we also need a change to `Makefile` to add something like...

```
edpm_deploy_prep: export DEPLOYMENT_NAME=${SOME_ENV_VAR}
```

...for this to work?

Comment (fao89) [reply scripts/gen-edpm-kustomize.sh:56]: I tested locally and it works:
```
dpadev@osp-dev-05:~/install_yamls$ DEPLOYMENT_NAME="testing-value" make edpm_deploy_prep
bash scripts/gen-namespace.sh
+ '[' -z /home/dpadev/install_yamls/out ']'
+ '[' -z openstack ']'
+ OUT_DIR=/home/dpadev/install_yamls/out/openstack
+ '[' '!' -d /home/dpadev/install_yamls/out/openstack ']'
+ cat
oc apply -f /home/dpadev/install_yamls/out/openstack/namespace.yaml
namespace/openstack unchanged
timeout 300s bash -c "while ! (oc get project.v1.project.openshift.io openstack); do sleep 1; done"
NAME        DISPLAY NAME   STATUS
openstack                  Active
oc project openstack
Already on project "openstack" on server "https://api.crc.testing:6443".
oc kustomize /home/dpadev/install_yamls/out/openstack/dataplane/cr | oc delete --ignore-not-found=true -f -
rm -Rf /home/dpadev/install_yamls/out/operator/dataplane-operator /home/dpadev/install_yamls/out/openstack/dataplane/cr
mkdir -p /home/dpadev/install_yamls/out/operator /home/dpadev/install_yamls/out/openstack-operators/dataplane/op /home/dpadev/install_yamls/out/openstack/dataplane/cr
bash scripts/clone-operator-repo.sh
Cloning repo: git clone  -b main https://github.com/openstack-k8s-operators/dataplane-operator.git dataplane-operator
Cloning into 'dataplane-operator'...
remote: Enumerating objects: 11151, done.
remote: Counting objects: 100% (11151/11151), done.
remote: Compressing objects: 100% (3517/3517), done.
remote: Total 11151 (delta 7643), reused 10721 (delta 7409), pack-reused 0
Receiving objects: 100% (11151/11151), 3.11 MiB | 3.85 MiB/s, done.
Resolving deltas: 100% (7643/7643), done.
cp /home/dpadev/install_yamls/out/operator/dataplane-operator/config/services/dataplane_v1beta1_openstackdataplaneservice_nova.yaml /home/dpadev/install_yamls/out/openstack/dataplane/cr
cp /dev/null /home/dpadev/install_yamls/out/openstack/dataplane/cr/25-nova-extra.conf
oc kustomize --load-restrictor LoadRestrictionsNone /home/dpadev/install_yamls/out/operator/dataplane-operator/examples/preprovisioned > /home/dpadev/install_yamls/out/openstack/dataplane/cr/dataplane.yaml
bash scripts/gen-edpm-kustomize.sh
+++ dirname scripts/gen-edpm-kustomize.sh
++ cd scripts
++ pwd -P
+ SCRIPTPATH=/home/dpadev/install_yamls/scripts
+ . /home/dpadev/install_yamls/scripts/common.sh --source-only
++ set -e
+ '[' -z openstack ']'
+ '[' -z OpenStackDataPlaneNodeSet ']'
+ '[' -z /home/dpadev/install_yamls/out/openstack/dataplane/cr ']'
+ NAME=openstackdataplanenodeset
+ '[' '!' -d /home/dpadev/install_yamls/out/openstack/dataplane/cr ']'
+ pushd /home/dpadev/install_yamls/out/openstack/dataplane/cr
~/install_yamls/out/openstack/dataplane/cr ~/install_yamls
+ cat
+ '[' -n '' ']'
+ cat
+ oc get pvc ansible-ee-logs -n openstack
+ cat
+ '[' 1 -gt 1 ']'
+ . /home/dpadev/install_yamls/scripts/gen-nova-custom-dataplane-service.sh
++ set -e
++ '[' compute == compute ']'
++ cat
++ cat
++ yq -i '(.spec.services[] | select(. == "nova")) |= "nova-custom"' dataplane.yaml
+ kustomization_add_resources
+ echo merge config dir
merge config dir
++ find . -type f -name '*.yaml'
++ grep -v kustomization
+ yamls='./dataplane_v1beta1_openstackdataplaneservice_nova.yaml
./dataplane.yaml'
+ for y in ${yamls[@]}
+ kustomize edit add resource ./dataplane_v1beta1_openstackdataplaneservice_nova.yaml
+ for y in ${yamls[@]}
+ kustomize edit add resource ./dataplane.yaml
+ popd
~/install_yamls
make edpm_deploy_generate_keys
make[1]: Entering directory '/home/dpadev/install_yamls'
devsetup/scripts/gen-ansibleee-ssh-key.sh
+++ dirname devsetup/scripts/gen-ansibleee-ssh-key.sh
++ cd devsetup/scripts
++ pwd -P
+ SCRIPTPATH=/home/dpadev/install_yamls/devsetup/scripts
+ NAMESPACE=openstack
+ METADATA_NAME=dataplane-ansible-ssh-private-key-secret
+ OUTPUT_DIR=../../out/edpm
+ SSH_ALGORITHM=rsa
+ SSH_KEY_FILE=ansibleee-ssh-key-id_rsa
+ SSH_KEY_SIZE=4096
+ pushd /home/dpadev/install_yamls/devsetup/scripts
~/install_yamls/devsetup/scripts ~/install_yamls
+ '[' '!' -d ../../out/edpm ']'
+ pushd ../../out/edpm
~/install_yamls/out/edpm ~/install_yamls/devsetup/scripts ~/install_yamls
+ oc get secret dataplane-ansible-ssh-private-key-secret -n openstack
+ echo 'Secret dataplane-ansible-ssh-private-key-secret already exists.'
Secret dataplane-ansible-ssh-private-key-secret already exists.
+ echo 'Delete it first to recreate:'
Delete it first to recreate:
+ echo 'oc delete secret dataplane-ansible-ssh-private-key-secret'
oc delete secret dataplane-ansible-ssh-private-key-secret
+ exit 0
bash scripts/gen-edpm-nova-migration-ssh-key.sh
+ oc get secret nova-migration-ssh-key -n openstack
NAME                     TYPE                     DATA   AGE
nova-migration-ssh-key   kubernetes.io/ssh-auth   2      89s
make[1]: Leaving directory '/home/dpadev/install_yamls'
oc apply -f devsetup/edpm/services
openstackdataplaneservice.dataplane.openstack.org/repo-setup configured
dpadev@osp-dev-05:~/install_yamls$ oc kustomize out/openstack/dataplane/cr/ | grep "testing-value"
  name: testing-value
```

Comment (fao89) [reply scripts/gen-edpm-kustomize.sh:56]:
```
dpadev@osp-dev-05:~/install_yamls$ oc kustomize out/openstack/dataplane/cr/ | yq 'select(.kind == "OpenStackDataPlaneDeployment")'
apiVersion: dataplane.openstack.org/v1beta1
kind: OpenStackDataPlaneDeployment
metadata:
  labels:
    created-by: install_yamls
  name: testing-value
  namespace: openstack
spec:
  nodeSets:
    - openstack-edpm-ipam
  services: []
```

Comment (abays) [reply scripts/gen-edpm-kustomize.sh:56]: Ack

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**